networkx/tests/test_convert_geopandas.py

```diff
@@ -0,0 +1,85 @@
+import os
+import pytest
+import networkx as nx
+
+gpd = pytest.importorskip("geopandas")
+
+dirname = os.path.dirname(__file__)
+
+
+def test_from_geopandas_edgelist_undirected():
+
+    gdf = gpd.GeoDataFrame.from_file(os.path.join(dirname, "test_shp.shp"))
+    graph_from_shp = nx.from_geopandas_edgelist(gdf, create_using=nx.Graph)
+    assert isinstance(graph_from_shp, nx.Graph)
+
+
+def test_from_geopandas_edgelist_directed():
+
+    gdf = gpd.GeoDataFrame.from_file(os.path.join(dirname, "test_shp.shp"))
+    graph_from_shp = nx.from_geopandas_edgelist(gdf, create_using=nx.DiGraph)
+    assert isinstance(graph_from_shp, nx.DiGraph)
+
+
+def test_from_geopandas_edgelist_undirected_with_attrs():
+
+    gdf = gpd.GeoDataFrame.from_file(os.path.join(dirname, "test_shp.shp"))
+    graph_from_shp = nx.from_geopandas_edgelist(
+        gdf, create_using=nx.Graph, edge_attr=gdf.columns.to_list()
+    )
+    failed = False
+    for u, v, d in graph_from_shp.edges(data=True):
+        if set(d.keys()) != {"id", "attr_text", "attr_float", "geometry", "attr_int"}:
+            failed = True
```

Comment:
You can put the assert statement in the loop. It will fail upon the first non-equal case just as this code does.
Similarly for the directed case below.